docs/other/overriding-configurations.md

```diff
@@ -1,4 +1,14 @@
-## Overriding variables in `config.yml` with a `local.config.yml`
+## Overriding Drupal VM's `default.config.yml` with `config.yml`
+
+If available, Drupal VM will load a `config.yml` where you can override any of the defaults set in `default.config.yml`. Commonly this is a copy of `default.config.yml` with the values tweaked to your own project. For an easier upgrade path you would only set the values you are actually overriding.
+
+```yaml
+vagrant_box: geerlingguy/centos7
+vagrant_hostname: my-custom-site.dev
```

Comment:
I verified that this particular example works. Ansible will dynamically evaluate the variable when it reaches the apache role and the correct hostname will be used in all places.